cmd/oras/root/discover.go

```diff
@@ -144,29 +144,29 @@ func runDiscover(cmd *cobra.Command, opts *discoverOptions) error {
 			return err
 		}
 		for _, ref := range refs {
-			if err := handler.OnDiscovered(ref, desc); err != nil {
+			if err := handler.OnDiscovered(ref, desc, ctx, repo, opts.Platform); err != nil {
 				return err
 			}
 		}
 	}
 	return handler.OnCompleted()
 }
 
-func fetchAllReferrers(ctx context.Context, repo oras.ReadOnlyGraphTarget, desc ocispec.Descriptor, artifactType string, handler metadata.DiscoverHandler) error {
+func fetchAllReferrers(ctx context.Context, repo oras.ReadOnlyGraphTarget, desc ocispec.Descriptor, artifactType string, handler metadata.DiscoverHandler, platform option.Platform) error {
 	results, err := registry.Referrers(ctx, repo, desc, artifactType)
 	if err != nil {
 		return err
 	}
 
 	for _, r := range results {
-		if err := handler.OnDiscovered(r, desc); err != nil {
+		if err := handler.OnDiscovered(r, desc, ctx, repo, platform); err != nil {
```

Comment:
Not loving passing ctx in the middle here. I haven't looked everything over, but seems like this should be passing down a fetch method around here